src/openfl/display/DisplayObject.hx

```diff
@@ -1325,26 +1320,92 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable #if openf
 	}
 	
 	
-	public function __updateChildren (transformOnly:Bool):Void {
+	private function __updateRenderData ():Void {
 		
 		var renderParent = __renderParent != null ? __renderParent : parent;
+		if (__isMask && renderParent == null) renderParent = __maskTarget;
 		__renderable = (visible && __scaleX != 0 && __scaleY != 0 && !__isMask && (renderParent == null || !renderParent.__isMask));
-		__worldAlpha = alpha;
-		__worldBlendMode = blendMode;
 		
-		if (__transformDirty) {
+		__updateTransforms ();
+		
+		if (__supportDOM) {
+				
+				__renderTransformChanged = !__renderTransform.equals (__renderTransformCache);
+				
+				if (__renderTransformCache == null) {
+					
+					__renderTransformCache = __renderTransform.clone ();
+					
+				} else {
+					
+					__renderTransformCache.copyFrom (__renderTransform);
+					
+				}
+				
+			}
 			
-			__transformDirty = false;
+			if (!__worldColorTransform.__equals (transform.colorTransform)) {
+				
+				__worldColorTransform = transform.colorTransform.__clone ();
+				
+			}
```

Comment:
I think this check is to prevent cloning the color transform on each update, but this only works well if there is no parent. Overall, this again is the discussion about different dirty flags and to update just what is dirty. Not that we have to change this now, I just wrote it down to leave my thoughts somewhere `;)`